.github/workflows/obs-upload.sh

```diff
@@ -0,0 +1,189 @@
+#!/bin/sh
+set -e
+
+RELEASES='rhel:7 rhel:8 fedora:32 fedora:33'
+
+for i in git curl xmlstarlet rpmspec
+do
+    if test -z "$(which "$i" || true)"
+    then
+        echo "The $i binary could not be found. Aborting."
+        exit 1
+    fi
+done
+
+BASE="$(git rev-parse --show-toplevel 2> /dev/null)"
+SPECFILE="${BASE}/ungoogled-chromium.spec"
+
+if test -z "${BASE}"
+then
+    echo "BASE directory could not be determined. Aborting."
+    exit 1
+fi
+
+if test ! -f "${SPECFILE}"
+then
+    echo "${SPECFILE} must exist and be a regular file. Aborting."
+    exit 1
+fi
+
+for i in OBS_API_USERNAME OBS_API_PASSWORD
+do
+    if test -z "$(eval echo \$${i})"
+    then
+        echo "$i is not in the environment. Aborting."
+        exit 1
+    fi
+done
+
+PROJECT="${OBS_API_PROJECT:-home:${OBS_API_USERNAME}}"
+
+curl()
+{
+    for i in `seq 1 5`
+    do
+        command curl -sS -K - "${@}" << EOF
+user="${OBS_API_USERNAME}:${OBS_API_PASSWORD}"
+EOF
+        test "${?}" -eq 0 && return 0
+        sleep 30s
+    done
+    return 1
+}
+
+get_type()
+{
+    local TAG
+    local TYPE
+
+    TAG="$(git describe --tags --exact-match 2> /dev/null || true)"
+
+    if test -z "${TAG}"
+    then
+        TYPE='development'
+    else
+        TYPE='production'
+    fi
+
+    echo "${TYPE}"
+}
+
+generate_obs()
+{
+    local ROOT="${1}"
+    local release
+    local version
+    local filename
+    local url
+    local protocol
+    local host
+    local path
+    local i
+
+    sed -r '1c%global obs 1' "${SPECFILE}" > "${ROOT}/ungoogled-chromium.spec"
+    SPECFILE="${ROOT}/ungoogled-chromium.spec"
+
+    echo -n > urls
+
+    for i in ${RELEASES}
+    do
+        release="${i%:*}"
+        version="${i#*:}"
+        rpmspec -E "%global ${release} ${version}" -P "${SPECFILE}" | sed -n -r '/^Source[0-9]+:\s*https?:/s/Source[0-9]+:\s*//p' >> "${ROOT}/urls"
+    done
+
+    printf '<services>\n' > "${ROOT}/_service"
+    sort -u < "${ROOT}/urls" | while read url
+    do
+        protocol="$(echo "${url}" | cut -d / -f 1 | cut -d : -f 1)"
+        host="$(echo "${url}" | cut -d / -f 3)"
+        path="$(echo "${url}" | cut -d / -f 4-)"
+        printf '%s<service name="download_url">\n' '    ' >> "${ROOT}/_service"
+        printf '%s<param name="protocol">%s</param>\n' '        ' "${protocol}" >> "${ROOT}/_service"
+        printf '%s<param name="host">%s</param>\n' '        ' "${host}" >> "${ROOT}/_service"
+        printf '%s<param name="path">%s</param>\n' '        ' "${path}" >> "${ROOT}/_service"
+        printf '%s</service>\n' '    ' >> "${ROOT}/_service"
+    done
+    printf '</services>\n' >> "${ROOT}/_service"
+
+    rm -f "${ROOT}/urls"
+
+    cat > "${ROOT}/_constraints" << 'EOF'
+<constraints>
+    <hardware>
+        <disk>
+            <size unit="G">16</size>
+        </disk>
+        <memory>
+            <size unit="G">8</size>
```

Comment:
Did you tested actual building this? Last time I tried I needed a much larger memory size.

Reply:
It's derived from what I was using for the other distributions but I am beginning to think I need to merge them all into the project config file. It does work for CentOS 7 but I keep getting OOMs for Fedora.

Reply:
Yeah for Fedora I eventually used 32G memory, so I think it need to be much larger.

Reply:
It's just strange as I just needed 24G to resolve it for the other x86_64 builds.